code/API_definitions/device-data-volume.yaml

```
@@ -0,0 +1,478 @@
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "406":
+          $ref: "#/components/responses/Generic406"
+        "429":
+          $ref: "#/components/responses/Generic429"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "501":
+          $ref: "#/components/responses/Generic501"
+        "502":
+          $ref: "#/components/responses/Generic502"
+        "503":
+          $ref: "#/components/responses/Generic503"
+        "504":
+          $ref: "#/components/responses/Generic504"
+
+  /check:
+    post:
+      tags:
+        - Device Data Volume
+      summary: "Check if the remaining data volume is above a certain threshold"
+      description: Check if the remaining data volume for a device is above a given threshold.
+      operationId: checkDataVolume
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CheckRequest"
+        required: true
+      responses:
+        "200":
+          description: Boolean response indicating if the remaining data volume exceeds the provided threshold
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                type: boolean
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "406":
+          $ref: "#/components/responses/Generic406"
+        "429":
+          $ref: "#/components/responses/Generic429"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "501":
+          $ref: "#/components/responses/Generic501"
+        "502":
+          $ref: "#/components/responses/Generic502"
+        "503":
+          $ref: "#/components/responses/Generic503"
+        "504":
+          $ref: "#/components/responses/Generic504"
+
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+  parameters:
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        type: string
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+  schemas:
+    DeviceDataVolumeResponse:
+      type: object
+      properties:
+        dataVolumeCategory:
+          type: string
+          description: Estimated remaining data volume category
+          enum:
+            - "<200MB"
+            - "<1GB"
+            - "<5GB"
+            - ">=5GB"
```

I think MiB and GiB are meant here. MB means 10^6 bytes and GB means 10^9 bytes, whereas data volumes are usually measured as a power of 2.
```suggestion
          description: Estimated remaining data volume category, where "MiB" is 2^20 bytes and "GiB" is 2^30 bytes
          enum:
            - "<200MiB"
            - "<1GiB"
            - "<5GiB"
            - ">=5GiB"
```